src/data_access/FirebaseUserDataAccessObject.java

```diff
@@ -0,0 +1,38 @@
+package data_access;
+
+import entity.User;
+import java.io.IOException;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import org.json.JSONException;
+import use_case.login.LoginUserDataAccessInterface;
+import use_case.signup.SignupUserDataAccessInterface;
+
+public class FirebaseUserDataAccessObject
+    implements SignupUserDataAccessInterface, LoginUserDataAccessInterface {
+  static OkHttpClient client = new OkHttpClient();
+
+  @Override
+  public User get(String username) {
+    Request request =
+        new Request.Builder()
+            .url("https://csc207-830a5-default-rtdb.firebaseio.com/users/" + username + ".json")
```

Comment:
This url shouldn't be hard-coded, maybe we should store this in a config file or env file?
This article (https://dev.to/msc29/practical-clean-architecture-in-typescript-rust-python-3a6d) gives an example.
Note that this url appears in many places throughout the code but I just flagged it here.